### ISSUE TITLE
in `android.intentFilters`, `pathAdvancedPattern` is rejected by JSON schema

### ISSUE THREAD

**Issue by billinghamj** [expo/expo#26116]
### Summary

`pathAdvancedPattern` is one of the fields allowed by Android for intent filters - https://developer.android.com/guide/topics/manifest/data-element#path - introduced in API 31

When it is used in the `android.intentFilters` config section, it mostly works fine - there's no typing errors, it is generated correctly, etc. But when running `expo-cli doctor`, it's rejected

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 19.9.0 - /opt/homebrew/bin/node
      Yarn: 3.5.0 - /opt/homebrew/bin/yarn
      npm: 9.6.3 - /opt/homebrew/bin/npm
    Managers:
      CocoaPods: 1.12.0 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Xcode: 14.3/14E222b - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 3.10.2
      expo-cli: 6.3.7
    Expo Workflow: managed
```

### Please specify your device/emulator/simulator platform, model and version

Android 13, no specific device

### Error output

```
Error: Problems validating fields in app.json. Learn more: https://docs.expo.dev/workflow/configuration/
 • Field: android/intentFilters/0/data - must be object.
 • Field: android/intentFilters/0/data/1 - should NOT have additional property 'pathAdvancedPattern'.
 • Field: android/intentFilters/0/data - must match a schema in anyOf.
```

### Reproducible demo or steps to reproduce from a blank project

`app.config.ts`:

```typescript
import { ConfigContext, ExpoConfig } from '@expo/config';

export default (context: ConfigContext): ExpoConfig => {
	return {
		...context.config,
		version: '1.0.0',
		android: {
			package: 'com.jamesbillingham.testapp',
			intentFilters: [
				{
					autoVerify: true,
					action: 'VIEW',
					category: ['BROWSABLE', 'DEFAULT'],
					data: {
						scheme: 'https',
						host: 'testapp.jamesbillingham.com',
						pathAdvancedPattern: '/[^.~_-].*',
					},
				},
			],
		},
		plugins: [
			['expo-build-properties', {
				android: {
					minSdkVersion: 31,
					compileSdkVersion: 33,
					targetSdkVersion: 33,
					buildToolsVersion: '33.0.1',
				},
			}],
		],
	};
};
```

**Comment by DenysMoro** [comment]
No stale

**Comment by billinghamj** [comment]
Not stale

**Comment by billinghamj** [comment]
Not stale 🙄

**Comment by billinghamj** [comment]
:|

**Comment by metehandemir** [comment]
We also need this feature too. pathPattern doesn’t suffice for our use case.

**Comment by danger9224** [comment]
This is still an issue, even when using TypeScript in CNG plugins. The `AndroidIntentFiltersData` interface is missing the pathAdvancedPattern field, even though providing it still works as expected. So I think resolving this is a matter of adding the missing type no?

**Comment by mckelvey** [comment]
Oh please please please, let me, let me, let me get what I want... this time... (updated JSON schema)...

https://www.youtube.com/watch?v=t0co3CgoMMU

**Comment by fbaiodias** [comment]
Hey @byCedric, 

Tagging you here again as I saw that [you were able to fix a JSON schema issue a while ago](https://github.com/expo/expo/issues/16861). 

Any change you could also fix this one (or at least reopen the issue)?

From my understanding, all you would need to do is to add the following right next to were you have `pathPattern` in the `AndroidIntentFiltersData` object: 

```json
{
  "pathAdvancedPattern": {
    "description": "Advanced pattern for paths, available in API 31+ (Android 12 and later),
supports regular expressions",
    "type": "string"
  }
}
```

**Comment by mvincentong** [comment]
I rechecked this against current `main`, and `pathAdvancedPattern` support appears to have landed via #45718:

https://github.com/expo/expo/pull/45718

Current evidence:

- `packages/@expo/config-types/src/ExpoConfig.ts` includes `pathAdvancedPattern?: string` on `AndroidIntentFiltersData`.
- `packages/@expo/config-plugins/src/android/Manifest.ts` includes the manifest attribute type `'android:pathAdvancedPattern'?: string`.
- `docs/public/static/schemas/unversioned/app-config-schema.json` and `docs/public/static/schemas/v56.0.0/app-config-schema.json` both include the `pathAdvancedPattern` schema entry with the Android 12 / API 31 note.
- The `@expo/config-plugins` changelog entry for #45718 lists `pathAdvancedPattern` among the missing intent-filter type fields that were added.

So current schema/type validation should no longer reject this field.
